aiohttp/web_request.py

```diff
@@ -510,8 +511,26 @@ def text(self):
         return bytes_body.decode(encoding)
 
     @asyncio.coroutine
-    def json(self, *, loads=json.loads):
-        """Return BODY as JSON."""
+    def json(self, *, loads=json.loads, allowed_type=None):
+        """ Return BODY as JSON. If the `Content-Type` does not equal
+        allowed_type, then the request is rejected.
+        """
+
+        if allowed_type is None:  # do not check the content_type
+            pass
+        elif isinstance(allowed_type, str):
+            # parse the content_type
+            content_type = self.content_type
+            (mime_type, subtype, suffix, _) = parse_mimetype(content_type)
+            content_type = mime_type + '/' + subtype
+            if suffix != '':
+                content_type += '+' + suffix
+
+            if allowed_type != content_type:
+                raise HTTPNotAcceptable()
```

Comment:
Also, I'm not sure if this concrete exception should be raised.

Reply:
What would be wrong with just 

```python
def json(self, *, loads=json.loads, allowed_type='application/json'):
    if allowed_type and not self.content_type == allow_type:
        raise HTTPNotAcceptable(text='invalid content-type: "{}" not "{}".format(self.content_type, allowed_type)
    ...
```

?

If users really want more sophisticated behaviour than this they can add their own logic, there's nothing very complicated here that couldn't be replicated in a custom coroutine.